### PR TITLE
Handle front end through jsbundling-rails

### PR DESCRIPTION
**Pull request by Janell-Huyck**
This PR updates the treatment app from using Sprockets to using jsbundling-rails to handle the front end.  Documentation about this PR is located on [the wiki page.](https://github.com/uclibs/treatment_database/wiki/Upgrading-from-Sprockets-to-jsbundling%E2%80%90rails).

### CODE REVIEW

**Comment by Janell-Huyck** [comment]
Hold - not correctly deploying to libappstest.  QA does deploy with ruby version 3.3.1.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -14,4 +14,36 @@ def user_display_name(id)
   rescue ActiveRecord::RecordNotFound
     "User not found (ID: #{id})"
   end
+
+  # Creates and returns the path for an image being handled by webpack
+  # Use like: <%= image_tag webpack_image_path('delete.png'), class: 'delete-icon', alt: 'Delete' %>
+  # Exact image_name to be passed in can be determined from the manifest.json file in the public/build directory
+  def webpack_image_path(image_name)
+    # Determine the base path based on the environment
+    base_path = Rails.env.production? ? '/treatment_database/build' : '/build'
```
[the remark this answers is not in the document]
Comment (Janell-Huyck) [reply app/helpers/application_helper.rb:23]: Works well, thanks!

File: config/application.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
 require_relative 'boot'
 
 require 'rails/all'
+require 'dotenv-rails'
```
Comment (hortongn) [inline config/application.rb:6]: Just curious, do you know why dotenv-rails had to be required when we haven't done that in the past?

File: app/helpers/application_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (hortongn) [reply app/helpers/application_helper.rb:23]: I was suggesting you could use RAILS_RELATIVE_URL_ROOT for constructing your base_path here:
https://github.com/uclibs/treatment_database/blob/935763ca8cb71f07a93c59ad8cc1855777dfad4f/app/helpers/application_helper.rb#L23

File: config/application.rb (hunk 2 above)
Comment (Janell-Huyck) [reply config/application.rb:6]: I think that happened when I was struggling to get Rails to recognize the environment.  I'll try taking it out and seeing what happens...

Comment (Janell-Huyck) [reply config/application.rb:6]: Screenshot 2024-07-02 at 3 47 44 PM

It's required in application.rb, and I'm not getting the app to start without it.

File: app/helpers/application_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Janell-Huyck) [reply app/helpers/application_helper.rb]: For some reason I just thought of using it in the webpack config file.  Got it!

**Review by hortongn — APPROVED** [review]
Impressive work!  I know it was a lot of sweat and tears, but this and the documentation you created will be very helpful for our other projects as well.  merging

**Pull request merged**